pkg/adaptor/cloud/libvirt/libvirt.go

```diff
@@ -116,55 +128,201 @@ func uploadIso(isoFile string, isoVolName string, libvirtClient *libvirtClient)
 
 }
 
-func CreateDomain(ctx context.Context, libvirtClient *libvirtClient, v *vmConfig) (result *createDomainOutput, err error) {
+func getGuestForArchType(caps *libvirtxml.Caps, arch string, virttype string) (*libvirtxml.CapsGuest, error) {
+	for _, guest := range caps.Guests {
+		if guest.Arch.Name == arch && guest.OSType == virttype {
+			return &guest, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find any guests for architecture type %s/%s", virttype, arch)
+}
 
-	v.cpu = uint(2)
-	v.mem = uint(8)
-	v.rootDiskSize = uint64(10)
+func getHostCapabilities(client *libvirtClient) (*libvirtxml.Caps, error) {
```

Comment:
Can this method and some others which are only required to be called once be executed when the CAA starts and populates the `libvirtClient` struct ?
May be here https://github.com/confidential-containers/cloud-api-adaptor/blob/main/pkg/adaptor/cloud/libvirt/libvirt.go#L360 ?

Reply:
@bpradipt yes, absolutely possible. I have updated the code.